src/base/binding.coffee

```diff
@@ -64,11 +64,23 @@ qdInternal.binding = {
             # pass the real dom node to the binding function as the $element value
             return bindingFunction(bindingContext, qdInternal.dom.unwrap(domNode))
         catch err
+            ancestorPath = @_getNodeAncestry(domNode)
             error = new QuickdrawError("'#{err.message}' in binding '#{domNode.getAttribute(qd.getConfig('bindingAttribute'))}'", err)
             error.setBindingContext(bindingContext)
+            error.setAncestorPath(ancestorPath)
             qdInternal.errors.throw(error)
             return null
 
+    # Gets the tagName and className for all nodes in the ancestry chain of this VirtualDomNode
+    # @param [VirtualDomNode] domNode the node to calculate the ancestry chain of
+    # @return [String] the ancestry chain of the provide domNode
+    _getNodeAncestry : (domNode) ->
+        ancestorPath = []
+        while domNode?
+            ancestorPath.push(domNode.getProperty('tagName').toLowerCase() + "." + domNode.getProperty('className').replace(" ", "."))
```

Comment:
Any chance of preferring the ID of the DOM node (e.g. `div#greeting`), then falling back to the set of classes (`div.headline.highlighted.derp`) if it doesn’t exist? I suspect having the ID would be really handy when it’s available, but totally understand if we’re trying to keep this as simple as possible.

Reply:
That's a great idea! As an extension, I'm wondering if it's beneficial to print both the id and class names...

Reply:
I'm in favor of having both. `div#greeting.headline.highlighted.derp`

It would also be possible not to serialize all this; such as some dictionary like

``` javascript
{
  classes: [ 'foo', 'bar' ],
  id: 'cheese',
  tag: 'div'
}
```

But in general QD tries to avoid object allocations, so what you have here is probably more consistent internally.

Reply:
Silly question but pendantically should the node itself be included in its list of ancestors?  Also, returning the list in order of the closest to farthest seems consistent with a few other APIs, but those seemed tailored to traversing, while this seems more informational for reading.  Would the opposite order make more sense?  Or is the function name ambiguous?

Reply:
Yeah I was struggling to come up with a good name for it because I didn't want to overload the meaning of other things (e.g. using the word context), I'm open to suggestions

Good callout on the ordering of the array, I'll reverse it